plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/visitors/NoSelfChecker.java

```diff
@@ -176,6 +178,28 @@ public void beforeFunctionDef(FunctionDef node) {
         scope.push(Scope.SCOPE_TYPE_METHOD);
     }
 
+    private static boolean hasInterfaceInBase(FunctionDef node) {
+        try {
+            ClassDef parent = (ClassDef) node.parent;
+            for (exprType base : parent.bases) {
+                String rep = null;
+                if (base instanceof Str) {
+                    Str str = (Str) base;
+                    rep = str.s;
+                } else {
+                    rep = NodeUtils.getFullRepresentationString(base);
```

Comment:
This isn't the ideal approach, we should actually follow the import at the class definition (see the other comment related to this).